src/gui/versionwidget.cpp

```diff
@@ -43,74 +44,101 @@ static const QString versionLink = "versionLink";
-    _versionNumberLabel = new QLabel(this);
-    _versionNumberLabel->setContextMenuPolicy(Qt::PreventContextMenu);
-    _versionNumberLabel->setText(versionNumberLinkText);
-    versionVBox->addWidget(_versionNumberLabel);
-
-    const auto copyrightLabel = new QLabel(QString("Copyright %1").arg(APPLICATION_VENDOR));
-    copyrightLabel->setObjectName("description");
-    versionVBox->addWidget(copyrightLabel);
-
-    _updateButton = new QPushButton(this);
-    _updateButton->setObjectName("defaultbutton");
-    _updateButton->setFlat(true);
-    versionBox->addWidget(_updateButton);
+    initVersionInfoBloc(prefBloc);
+    prefBloc->addSeparator();
+    initBetaBloc(prefBloc);
 
     refresh();
 
-    // TODO : add it back later (version 3.6.8 or 4.0)
-    // connect(_prodButton, &QRadioButton::clicked, this, &VersionWidget::onChannelButtonClicked);
-    // connect(_betaButton, &QRadioButton::clicked, this, &VersionWidget::onChannelButtonClicked);
-    // connect(_internalButton, &QRadioButton::clicked, this, &VersionWidget::onChannelButtonClicked);
     connect(_updateStatusLabel, &QLabel::linkActivated, this, &VersionWidget::onLinkActivated);
     connect(_versionNumberLabel, &QLabel::linkActivated, this, &VersionWidget::onLinkActivated);
     connect(_showReleaseNotesLabel, &QLabel::linkActivated, this, &VersionWidget::onLinkActivated);
     connect(_updateButton, &QPushButton::clicked, this, &VersionWidget::onUpdateButtonClicked);
+    connect(_joinBetaButton, &QPushButton::clicked, this, &VersionWidget::onJoinBetaButtonClicked);
+}
+
+void VersionWidget::refresh(const bool isStaff) {
+    _isStaff = isStaff;
+    refresh();
+}
+
+void VersionWidget::showAboutDialog() {
+    EnableStateHolder _(this);
+    AboutDialog dialog(this);
+    dialog.execAndMoveToCenter(GuiUtility::getTopLevelWidget(this));
+}
+
+void VersionWidget::showReleaseNotes() const {
+    QString os;
+#if defined(__APPLE__)
+    os = "macos";
+#elif defined(_WIN32)
+    os = "win";
+#else
+    os = "linux";
+#endif
+
+    VersionInfo versionInfo;
+    GuiRequests::versionInfo(versionInfo);
+
+    const Language &appLanguage = ParametersCache::instance()->parametersInfo().language();
+    QString languageCode = CommonUtility::languageCode(appLanguage);
+    if (languageCode.isEmpty()) languageCode = "en";
+    QDesktopServices::openUrl(
+            QUrl(QString("%1-%2-%3-%4.html")
+                         .arg(APPLICATION_STORAGE_URL, versionInfo.fullVersion().c_str(), os, languageCode.left(2))));
+}
+
+void VersionWidget::showDownloadPage() const {
+    QDesktopServices::openUrl(QUrl(APPLICATION_DOWNLOAD_URL));
+}
+
+void VersionWidget::onUpdateStateChanged(const UpdateState state) const {
+    refresh(state);
+}
+
+void VersionWidget::onLinkActivated(const QString &link) {
+    if (link == versionLink)
+        showAboutDialog();
+    else if (link == releaseNoteLink)
+        showReleaseNotes();
+    else if (link == downloadPageLink)
+        showDownloadPage();
```

Comment:
Log an error in case of mismatch? Assertion?

Reply:
Done d100d681f08c906c7e42fae7a357f693961ec2f0

Reply:
Done in d100d681f08c906c7e42fae7a357f693961ec2f0